services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/SqsBatchFunctions.java

```diff
@@ -0,0 +1,323 @@
+            // requests must have the same overrideConfiguration so it is sufficient to retrieve it from the first request.
+            Optional<AwsRequestOverrideConfiguration> overrideConfiguration = identifiedRequests.get(0)
+                                                                                                .message()
+                                                                                                .overrideConfiguration();
+            SendMessageBatchRequest batchRequest =
+                overrideConfiguration.map(overrideConfig -> SendMessageBatchRequest.builder()
+                                                                                   .queueUrl(destination)
+                                                                                   .overrideConfiguration(overrideConfig)
+                                                                                   .entries(entries)
+                                                                                   .build())
+                                     .orElse(SendMessageBatchRequest.builder()
+                                                                    .queueUrl(destination)
+                                                                    .entries(entries)
+                                                                    .build());
+            return CompletableFuture.supplyAsync(() -> client.sendMessageBatch(batchRequest), executor);
+        };
+    }
+
+    public static BatchResponseMapper<SendMessageBatchResponse, SendMessageResponse> sendMessageResponseMapper() {
+        return sendMessageBatchResponse -> {
+            List<IdentifiableMessage<SendMessageResponse>> mappedResponses = new ArrayList<>();
+            sendMessageBatchResponse.successful()
+                                    .forEach(batchResponseEntry -> {
+                                        String key = batchResponseEntry.id();
+                                        SendMessageResponse response = createSendMessageResponse(batchResponseEntry,
+                                                                                                 sendMessageBatchResponse);
+                                        mappedResponses.add(new IdentifiableMessage<>(key, response));
+                                    });
+            sendMessageBatchResponse.failed()
+                                    .forEach(batchResponseEntry -> {
+                                        String key = batchResponseEntry.id();
+                                        SendMessageResponse response = createSendMessageResponse(batchResponseEntry,
+                                                                                                 sendMessageBatchResponse);
+                                        mappedResponses.add(new IdentifiableMessage<>(key, response));
+                                    });
```

Comment:
Hmm, if it fails, shouldn't we return exception for that specific request? (sorry, missed this in my previous reviews). If this requires refactoring of the core, let's fix it in a separate PR

Reply:
Yeah we discussed wrapping the exception message inside a sendMessageResponse instead of actually returning an exception since it would involve refactoring the core (right now the core expects a list of sendMessageResponses to be returned so I would have to refactor the core to take a list of sendMessageResponses and exceptions instead).

Reply:
I see. We can probably use `List<Either<SendMesasgeResponse, Throwable>>` https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/waiters/WaiterExecutor.java#L73

Can we just use an empty SendMessageResponse(without md5) as part of this PR?

Reply:
Returning empty sendMessageResponse for now.